### PR TITLE
Webserver: Further Sanitize values passed to origin param

### PR DESCRIPTION
**Pull request by kaxil**
Follow-up of https://github.com/apache/airflow/pull/10334

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by dimberman — APPROVED** [review]

File: airflow/www/views.py [hunk 1]
```diff
@@ -108,8 +108,13 @@ def get_safe_url(url):
     valid_schemes = ['http', 'https', '']
     valid_netlocs = [request.host, '']
 
+    # Remove single quotes
+    url = url.replace("'", "")
```
Comment (ashb) [inline airflow/www/views.py]: What's this?

File: airflow/www/views.py [hunk 2]
```diff
@@ -108,8 +108,13 @@ def get_safe_url(url):
...
     parsed = urlparse(url)
 
+    query = parse_qsl(parsed.query)
+    url = parsed._replace(query=urlencode(query)).geturl()
```
Comment (ashb) [inline airflow/www/views.py:117]: Using a private method isn't great. Do we have to?

Comment (kaxil) [reply airflow/www/views.py:117]: It is NamedTuple's _replace method: https://docs.python.org/3/library/collections.html#collections.somenamedtuple._replace

Just returns a new instance of the named tuple replacing specified fields with new values

Comment (ashb) [reply airflow/www/views.py:117]: Right, so not private, just looks like it. Funsies.

File: airflow/www/views.py (hunk 1 above)
Comment (ashb) [inline airflow/www/views.py]:
```suggestion
```

File: airflow/www/views.py [hunk 3]
```diff
@@ -108,8 +108,13 @@ def get_safe_url(url):
     valid_schemes = ['http', 'https', '']
     valid_netlocs = [request.host, '']
 
+    # Remove single quotes
+    url = url.replace("'", "")
     parsed = urlparse(url)
 
+    query = parse_qsl(parsed.query)
```
Comment (ashb) [inline airflow/www/views.py]:
```suggestion
    query = parse_qsl(parsed.query, keep_blank_values=True)
```

File: tests/www/test_views.py [hunk 4]
```diff
@@ -3293,3 +3294,25 @@ def test_action_logging_post(self):
...
         )
+
+
+class TestHelperFunctions(unittest.TestCase):
+    @parameterized.expand(
+        [
+            ("http://google.com", "/home"),
+            (
+                "http://localhost:8080/trigger?dag_id=test_dag&origin=%2Ftree%3Fdag_id%test_dag';alert(33)//",
+                "http://localhost:8080/trigger?dag_id=test_dag&origin=%2Ftree%3Fdag_id%25test_dag",
```
Comment (ashb) [inline tests/www/test_views.py]:
```suggestion
                "http://localhost:8080/trigger?dag_id=test_dag&origin=%2Ftree%3Fdag_id%25test_dag&alert%2833%29%2F%2F=",
```

**Review by ashb — CHANGES_REQUESTED** [review]

File: tests/www/test_views.py (hunk 4 above)
Comment (kaxil) [reply tests/www/test_views.py]: fixed in https://github.com/apache/airflow/pull/12459/commits/7486153f451e4d2bb1c6fd9cbb5a63430157c99c

File: airflow/www/views.py (hunk 3 above)
Comment (kaxil) [reply airflow/www/views.py]: fixed in https://github.com/apache/airflow/pull/12459/commits/7486153f451e4d2bb1c6fd9cbb5a63430157c99c

File: airflow/www/views.py (hunk 1 above)
Comment (kaxil) [reply airflow/www/views.py]: fixed in https://github.com/apache/airflow/pull/12459/commits/7486153f451e4d2bb1c6fd9cbb5a63430157c99c

**Review by ashb — APPROVED** [review]

**Pull request merged**